pennylane/control_flow/while_loop.py

```diff
@@ -87,6 +95,78 @@ def loop_rx(x):
 
     >>> qml.qjit(circuit)(1.6)
     Array(-0.02919952, dtype=float64)
+
+    .. details::
+        :title: Usage Details
+
+        A dynamicly shaped array is an array whose shape depends on an abstract value. This is
+        an experimental jax mode that can be turned on with:
+
+        >>> jax.config.update("jax_dynamic_shapes", True)
+
+        Note that new dynamic dimensions cannot yet be created inside a loop.  Only things
+        that already have a dynamic dimension can have that dynamic dimension change.
+        For example, this is **not** a viable ``while_loop``, as ``x`` is initialized
+        with an array with a concrete size.
+
+        .. code-block:: python
+
+            def w():
+                @qml.while_loop(lambda i, x: i < 5)
+                def f(i, x):
+                    return i + 1, jax.numpy.append(x, i)
+
+                return f(0, jnp.array([]))
+
+        When working with dynamic shapes in a ``while_loop``, we have two possible
+        options. In ``allow_array_resizing=True`` treats every dynamic dimension as independent.
+
+        .. code-block:: python
+
+            @qml.while_loop(lambda a, b: jax.numpy.sum(a) < 10, allow_array_resizing=True)
+            def f(a, b):
+                return jax.numpy.hstack([a, b]), 2*b
+
+            def w(i0):
+                a0, b0 = jnp.ones(i0), jnp.ones(i0)
+                return f(a0, b0)
```

Comment:
```suggestion
            @qml.while_loop(lambda a, b: jax.numpy.sum(a) < 10, allow_array_resizing=True)
            def f(x, y):
                return jax.numpy.hstack([x, y]), 2*y

            def workflow(i0):
                x0, y0 = jnp.ones(i0), jnp.ones(i0)
                return f(x0, y0)
```
Changing to be the same as the example below.